### ISSUE TITLE
Is it supposed to look different than in Preview?

### ISSUE THREAD

**Issue by kladderadeng** [ckant/joplin-plugin-better-code-blocks#8]
Hi there,

First, Thank you very much for creating this plugin and sharing your work with us.
I make my first steps with Joplin and since there will be quite some code in my notes, I installed your plugin as one of the first ones. I would like to stay in the Markdown editor as long as possible, so inline Code formatting looked interesting to me.

But I am not sure if it is not working for me properly or if it is intended. I expected to have at least a similar style between the markdown editor and the preview, but most of my code (Powershell in this example) does not even get highlighted in the markdown editor. Here is an example:

![image](https://github.com/user-attachments/assets/03f96ca5-c774-4dd3-8339-f3981c69196d)

Also Autocomplete for Powershell does not seem to work, but then again: Maybe my understanding is incorrect. Autocomplete maybe does not need autocomplete while writing the code, but more autocomplete for the ```powersh<enter> thing?

**Comment by ckant** [comment]
> But I am not sure if it is not working for me properly or if it is intended. I expected to have at least a similar style between the markdown editor and the preview, but most of my code (Powershell in this example) does not even get highlighted in the markdown editor.

Yes it's working as intended. Joplin uses [a rather simple CodeMirror theme](https://github.com/laurent22/joplin/blob/2b43a9a4d667fe6b81bc97b66e0b3700688ec3cf/packages/editor/CodeMirror/theme.ts#L269-L306) to syntax highlight the editor and uses [a more comprehensive highlightjs theme](https://github.com/laurent22/joplin/blob/2b43a9a4d667fe6b81bc97b66e0b3700688ec3cf/packages/renderer/assets/highlight.js/atom-one-dark-reasonable.css) for the preview. CodeMirror has no out-of-the-box support for highlightjs, but [it might be possible to bridge the two with the right mapping](https://github.com/SINTEF/sourcegraph/blob/7e3ddab03e849394e3e92026307d0579b7ded8f6/client/shared/src/components/CodeMirrorEditor.ts#L246-L271) (no idea how well this works in practice).

Another thing you can use is [my Editor Themes plugin](https://joplinapp.org/plugins/plugin/com.ckant.joplin-plugin-editor-themes/?from-tab=themes), which adds some pre-built alternative themes to the editor. There aren't any that are one-to-one with the preview theme, but they offer more definitions for syntax highlighting than the default one. Here's one of the themes showing a similar snippet.

<img width="624" alt="image" src="https://github.com/user-attachments/assets/13df7a2e-85b7-4711-b2e5-c29e155b9234" />

Unfortunately, the highlightjs theme still looks more comprehensive.

> Also Autocomplete for Powershell does not seem to work, but then again: Maybe my understanding is incorrect. Autocomplete maybe does not need autocomplete while writing the code, but more autocomplete for the ```powersh thing?

Correct, it is autocompletion of the ```powersh part.

The editor does support autocompletion for *some* languages [like javascript](https://github.com/codemirror/lang-javascript/blob/b994686c0958af1dc782b406c65cdcad0d645316/src/complete.ts). It can be added with [this plugin](https://joplinapp.org/plugins/plugin/io.github.personalizedrefrigerator.codemirror6-settings/) and the `Enable autocompletion` setting.

However, autocompletion isn't supported for `powershell`. It only [has highlighting support](https://github.com/codemirror/legacy-modes/blob/92b908e4483b85d406bb64d648448f54680e4c53/mode/powershell.js).

**Comment by kladderadeng** [comment]
Thank you very much for your comprehensive response and your suggestions for other Plugins. I will have a look at your Themes, they look very interesting. So far I wrote a lot of notes since I opened this issue and get more and more familiar with what I can do. I really enjoy writing my documentations in Joplin and plugins like yours help a lot. Thanks again :-)
